tools/compiler/src/processor.rs

```diff
@@ -60,21 +60,53 @@ pub async fn generate_registry() -> AppResult<()> {
     Ok(())
 }
 
+/// Given `ibc_data` describing a channel and `source_asset` on the source chain,
+/// compute the metadata for the asset when it is transported along the channel onto a Penumbra chain.
+fn transport_metadata_along_channel(ibc_data: &IbcConfig, source_asset: Metadata) -> Metadata {
+    // The `Metadata` structure doesn't allow modifying the internals, so drop to raw proto data
+    let mut pb_metadata: pb::Metadata = source_asset.into();
+    tracing::debug!(?pb_metadata, "original");
+
+    let prefix_channel = |x: &mut String| {
+        *x = format!("transfer/{}/{}", ibc_data.ibc_channel, x);
+    };
+
+    // Prefix the channel to the base denom.
+    prefix_channel(&mut pb_metadata.base);
+    // Prefix the channel to the display denom.
+    prefix_channel(&mut pb_metadata.display);
```

Comment:
Do we indeed want to show `transfer/channel-0/usomo` in our UI if we believe it's the canonical one? I believe minifront is using display in a few places 🤔 . I suppose if we rely on symbol, this wouldn't be an issue anyway. Though, a native assets on two different connected chains could in theory conflict on symbol.

Reply:
Yeah I think we should use this as a forcing function to only ever use the symbol files in normie UI